File: frontend/src/components/fix.js
const fs = require('fs');
const path = 'c:/Users/This PC/Downloads/UniScout_1.5.1/UniScout_1.5/frontend/src/components/Dashboard.jsx';
let content = fs.readFileSync(path, 'utf8');

// The broken string looks like:
// border ${formData. ? 'bg-white border-blue-100' : 'bg-red-50 border-red-500'} 
// We want to replace it by looking at the value={formData.XYZ} prop that follows it!

content = content.replace(/border \$\{formData\. \? 'bg-white border-blue-100' : 'bg-red-50 border-red-500'\}\s+rounded-xl([^>]+)value=\{formData\.(\w+)\}/g, (match, middle, propName) => {
  return `border \$\{formData.${propName} ? 'bg-white border-blue-100' : 'bg-red-50 border-red-500'\} rounded-xl${middle}value={formData.${propName}}`;
});

fs.writeFileSync(path, content);
console.log('Fixed file');

File: frontend/src/components/fix.py
import re

path = 'c:/Users/This PC/Downloads/UniScout_1.5.1/UniScout_1.5/frontend/src/components/Dashboard.jsx'

with open(path, 'r', encoding='utf-8') as f:
    content = f.read()

def replacer(match):
    middle = match.group(1)
    prop = match.group(2)
    return f"border ${{formData.{prop} ? 'bg-white border-blue-100' : 'bg-red-50 border-red-500'}}{middle}value={{formData.{prop}}}"

pattern = r"border \$\{formData\. \? 'bg-white border-blue-100' : 'bg-red-50 border-red-500'\}(\s+rounded-xl[^>]+)value=\{formData\.(\w+)\}"

new_content = re.sub(pattern, replacer, content)

with open(path, 'w', encoding='utf-8') as f:
    f.write(new_content)

print("Fix completed")
